llvm/lib/Transforms/Utils/FixIrreducible.cpp

```diff
@@ -79,6 +79,53 @@
 // Limitation: The pass cannot handle switch statements and indirect
 //             branches. Both must be lowered to plain branches first.
 //
+// CallBr support: CallBr is handled as a more general branch instruction which
+// can have multiple successors. The pass redirects the edges to intermediate
+// target blocks that unconditionally branch to the original callbr target
+// blocks. This allows the control flow hub to know to which of the original
+// target blocks to jump to.
+// Example input CFG:
+//                        Entry (callbr)
+//                       /     \
+//                      v       v
+//                      H ----> B
+//                      ^      /|
+//                       `----' |
+//                              v
+//                             Exit
+//
+// becomes:
+//                        Entry (callbr)
+//                       /     \
+//                      v       v
+//                 target.H   target.B
+//                      |       |
+//                      v       v
+//                      H ----> B
+//                      ^      /|
+//                       `----' |
+//                              v
+//                             Exit
```

Comment:
What is the difference to what https://github.com/llvm/llvm-project/blob/main/llvm/lib/Transforms/Utils/BreakCriticalEdges.cpp is doing?

Reply:
Sounds exactly like what `SplitEdge` is doing. The only complicated case is the one of critical edges, otherwise `SplitEdge` is identical to splitting the predecessor or successor block. For normalzing the CFG, splitting may not be necessary except for critical edges.